src/test/java/com/cdancy/bitbucket/rest/features/InsightsApiMockTest.java

```diff
@@ -0,0 +1,478 @@
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+    public void testGetReportOnError() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/insight-report-error.json")).setResponseCode(404));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final String reportKey = qwertyKeyword;
+            final InsightReport report = api.getReport(projectKey, repoKey, commitHash, reportKey);
+            assertThat(report).isNotNull();
+            assertThat(report.errors()).isNotEmpty();
+
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword + "/" + reportKey);
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+    public void testCreateReport() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/insight-report.json")).setResponseCode(200));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final String reportKey = qwertyKeyword;
+            final InsightReportData reportData = InsightReportData.createPercentage("Code Coverage", (byte) 15);
+            final CreateInsightReport createInsightReport = CreateInsightReport.create("details",
+                                                                                       "http://example.com",
+                                                                                       "http://example.com/logourl",
+                                                                                       CreateInsightReport.RESULT.PASS,
+                                                                                       "reportTitle",
+                                                                                       "Bitbucket-rest",
+                                                                                       Collections.singletonList(reportData));
+            final InsightReport report = api.createReport(projectKey, repoKey, commitHash, reportKey, createInsightReport);
+            assertThat(report).isNotNull();
+            assertThat(report.key()).isEqualTo(reportKey);
+
+            assertSent(server,
+                       putMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword + "/" + reportKey);
```

Comment:
:warning: Line is longer than 140 characters (found 147). (#QR112:185#)-CHECKSTYLE_LineLengthCheck